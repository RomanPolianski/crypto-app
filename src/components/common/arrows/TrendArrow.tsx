/* eslint-disable react/require-default-props */
import classNames from 'classnames';
import { FC } from 'react';
import styles from './TrendArrow.module.scss';

interface TrendArrowProps {
  isRising: boolean;
  difference?: number;
}

export const TrendArrow: FC<TrendArrowProps> = ({
  isRising,
  difference,
}): JSX.Element => {
  return (
    <i
      className={classNames(
        styles.arrow,
        isRising ? styles.arrow__green : styles.arrow__red,
        difference === 0 ? styles.arrow__black : '',
        isRising ? styles.arrow__up : styles.arrow__down
      )}
    />
  );
};
