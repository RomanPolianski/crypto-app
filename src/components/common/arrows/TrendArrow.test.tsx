import { render } from '../../../../test-utils';
import { TrendArrow } from './TrendArrow';

describe('TrendArrow test', () => {
  it('TrendArrow snapshot down red', () => {
    const arrow = render(<TrendArrow isRising difference={-21} />);
    expect(arrow).toMatchSnapshot();
  });

  it('TrendArrow snapshot up green', () => {
    const arrow = render(<TrendArrow isRising difference={31} />);
    expect(arrow).toMatchSnapshot();
  });
});
