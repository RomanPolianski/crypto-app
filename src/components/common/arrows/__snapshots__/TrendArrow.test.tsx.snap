// Vitest Snapshot v1

exports[`TrendArrow test > TrendArrow snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <i
        class="_arrow_1gmpk_1 _arrow__green_1gmpk_21 _arrow__up_1gmpk_16"
      />
    </div>
  </body>,
  "container": <div>
    <i
      class="_arrow_1gmpk_1 _arrow__green_1gmpk_21 _arrow__up_1gmpk_16"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TrendArrow test > TrendArrow snapshot down red 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <i
        class="_arrow_1gmpk_1 _arrow__green_1gmpk_21 _arrow__up_1gmpk_16"
      />
    </div>
  </body>,
  "container": <div>
    <i
      class="_arrow_1gmpk_1 _arrow__green_1gmpk_21 _arrow__up_1gmpk_16"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`TrendArrow test > TrendArrow snapshot up green 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <i
        class="_arrow_1gmpk_1 _arrow__green_1gmpk_21 _arrow__up_1gmpk_16"
      />
    </div>
  </body>,
  "container": <div>
    <i
      class="_arrow_1gmpk_1 _arrow__green_1gmpk_21 _arrow__up_1gmpk_16"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
